### ISSUE TITLE
Help with nested stacks

### ISSUE THREAD

**Issue by cattz** [awslabs/aws-deployment-framework#149]
I think I'm not understanding the way to work with nested stacks.
From the User Guide:

> This allows us to specify nested stacks that are in the same repository as our main `template.yml` in our like so:
```
  MyStack:
    Type: "AWS::CloudFormation::Stack"
    Properties:
      TemplateURL: another_template.yml # file path to the nested stack template
```
> When the package_transform.sh command is executed, the file will be packaged up and uploaded to Amazon S3. Its TemplateURL key will be updated to point to the object in S3 and this will be a valid path when template.yml is executed in the deploy stages of your pipeline.

This is exactly what I'm trying to do, but facing some issues:
1. The `package_transform.sh` is not trying to package my templates, since it's greping for `Transform:` in the main template, but I have no transforms there.
I've "fixed" this in an extremely hacky way by adding a comment with that content. Worth noting that, even I added the `contains_transform: true` in the pipeline definition in the deployment map, that has not made any difference.
What is the proper way to resolve this? 
Do we need to add another grep for `TemplateUrl:`?
Should I use transform for including other templates in my main one? How?

2. Once I get the `sam package` command running, I get the below error in the build stage of the pipeline:
```
Unable to upload artifact ../../another_template.yml referenced by TemplateURL parameter of MyStack resource. 
TemplateURL parameter of MyStack resource is invalid. It must be a S3 URL or path to CloudFormation template file. Actual: /codebuild/output/src45678/src/another_template.yml
```
At this point I'm quite lost.
Would it make sense to add a full example of nested templates?

**Comment by cattz** [comment]
Ok, you can ignore the second point. That one was a 'human error', as in a typo in the template file name 😝

**Comment by thomasmcgannon** [comment]
Hi @cattz, I think it is a case of the package_transform being originally written to handle transforms and the logic to check for "Transform:" is just a sanity check to avoid packaging up when its not necessary due to human error.
I think your solution is quite right, we can remove the grep on Transform and then it is up to the user to only use `contains_transform: true` only when necessary.

So the steps to take are:
1. Remove the grep for "Transform:"
2. Ensure `contains_transform: true` (We do this because package_transform.sh will generate different template.yml files, and so the pipeline needs to be rebuilt to know to expect these different names)

Now when you run it it will package it all up, upload to S3 and replace the TemplateURL to the S3 URL.

If you'd like to create a PR with the change of removing the grep line then we can include that in a future versions.

Let me know how you go!

**Comment by cattz** [comment]
Hi @thomasmcgannon,
I have created a PR. Tested this with my setup and seems to be working as expected

**Comment by bundyfx** [comment]
Thanks @cattz its been merged into master for 2.0.0
